apps/epic-web/src/app/articles/[slug]/edit/page.tsx

```diff
@@ -0,0 +1,39 @@
+import {getArticle} from 'lib/articles'
+import {notFound} from 'next/navigation'
+import serializeMDX from '@skillrecordings/skill-lesson/markdown/serialize-mdx'
+import readingTime from 'reading-time'
+import MDX from '@skillrecordings/skill-lesson/markdown/mdx'
+import React from 'react'
+import {getServerAuthSession} from 'server/auth'
+import {getCurrentAbility} from '@skillrecordings/skill-lesson/utils/ability'
+
+export default async function ArticleEditPage({
+  params,
+}: {
+  params: {slug: string}
+}) {
+  const session = await getServerAuthSession()
+  const ability = getCurrentAbility({user: session?.user})
+  const article = await getArticle(params?.slug as string)
+
+  if (!article || !ability.can('create', 'Content')) {
+    notFound()
+  }
```

Comment:
we can load content and check abilities

because this is an edit route we can technically make everything below this a `use client` since we don't care about SEO